packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts

```diff
@@ -324,24 +324,33 @@ export class AwsCustomResource extends cdk.Construct implements iam.IGrantable {
     });
     this.grantPrincipal = provider.grantPrincipal;
 
+    // Create the policy statements for the custom resource function role, or use the user-provided ones
+    const statements = [];
     if (props.policy.statements.length !== 0) {
       // Use custom statements provided by the user
       for (const statement of props.policy.statements) {
         provider.addToRolePolicy(statement);
+        statements.push(statement);
       }
     } else {
       // Derive statements from AWS SDK calls
       for (const call of [props.onCreate, props.onUpdate, props.onDelete]) {
         if (call) {
-          provider.addToRolePolicy(new iam.PolicyStatement({
+          const statement = new iam.PolicyStatement({
             actions: [awsSdkToIamAction(call.service, call.action)],
             resources: props.policy.resources,
-          }));
+          });
+          provider.addToRolePolicy(statement);
```

Comment:
Is this still needed? Looking at the integ test expectations we see that permissions are now present two times: in a policy with the permissions of all custom resources and then again in separate policies per custom resources.

Reply:
I actually wanted to ask about this -- this will cause resource changes the next time someone with a custom resource updates their CDK. Would the update be safe, even if the end permissions are the same?